### ISSUE TITLE
Enhancement: [no-misused-promises] should include checks around synchronous `using` statements.

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10357]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-misused-promises/

### Description

I would like the rule to add a few checks for misused promises involved in explicit resource management code. This is split out from https://github.com/typescript-eslint/typescript-eslint/issues/10208#issuecomment-2439735239

### Fail

```TypeScript
// rule does not inspect variable for misused promise-returning functions
const a = {
    async [Symbol.dispose]() {}
} 
const b: Disposable = a;
//                    ~ <-- expected report here.


// `using` statement without explicit type
using c = {
    async [Symbol.dispose]() {}
//  ~~~~~~~~~~~~~~~~~~~~~~ <-- expected report here.
};

// combination of both of the above
const d = {
    async [Symbol.dispose]() {}
} 
using e = d;
//        ~ <-- expected report here.
```

### Pass

```TypeScript
// just a sync disposable
using a = {
    [Symbol.dispose]() {}
}

// proper use of await using
await using b = {
    async [Symbol.dispose]() {}
}
```

### Additional Info

Note that the upcoming strict-void-return aims to be a superset of no-misused-promises, so perhaps this check will make sense to add there also, see #9707
